pkg/apply/applier.go

```diff
@@ -370,19 +373,19 @@ func handleError(eventChannel chan event.Event, err error) {
 // validateNamespace returns true if all the objects in the passed
 // infos parameter have the same namespace; false otherwise. Ignores
 // cluster-scoped resources.
-func validateNamespace(infos []*resource.Info) bool {
-	currentNamespace := metav1.NamespaceNone
+func validateNamespace(inv *resource.Info, infos []*resource.Info) error {
+	invAcc, _ := meta.Accessor(inv.Object)
+	invNamespace := invAcc.GetNamespace()
+
 	for _, info := range infos {
-		// Ignore cluster-scoped resources.
-		if info.Namespaced() {
-			// If the current namespace has not been set--then set it.
-			if currentNamespace == metav1.NamespaceNone {
-				currentNamespace = info.Namespace
-			}
-			if currentNamespace != info.Namespace {
-				return false
+		acc, _ := meta.Accessor(info.Object)
+		gvk := info.Object.GetObjectKind().GroupVersionKind()
+		if gvk.Group == "" && gvk.Version == "v1" && gvk.Kind == "Namespace" &&
```

Comment:
nit: this if statement is a bit awkward. Another approach is to create a GVK of core/v1/Namespace, and compare that (for first three statements):

```
coreV1Namespace := schema.GroupVersionKind{
  Group: "",
  Version: "v1",
  Kind: "Namespace",
}

if gvk == coreV1Namespace && ...
```

Reply:
Fixed